services/site/src/components/Navigation.tsx

```diff
@@ -19,10 +22,16 @@ export interface NavigationProps {
 
 const Navigation: React.FC<React.PropsWithChildren<NavigationProps>> = ({ displayBreadcrumbs = true, children }) => {
   const { currentUser } = useCurrentUser();
-  const [logout] = useLogoutMutation({
+  const [triggerLogout] = useLogoutMutation({
     refetchQueries: [{ query: CurrentUserDocument }],
   });
   const userAccountModalApi = useUserAccountModalApi();
+  const router = useRouter();
+
+  const logout = async (): Promise<void> => {
+    await triggerLogout();
+    router.push("/");
```

Comment:
We now have to explicitly refresh the page, since the statically rendered content of the page for a logged in user doesn't match the content that should be shown for a anonymous user.